### DISCUSSION TITLE
Configure subdomain per request?

### DISCUSSION THREAD

**Discussion by SteveEdson** [saloonphp/saloon#95] [Q&A]
I have an API that uses tenancy, where is has a different subdomain per client, e.g. 

`https://{clientId}.host.com/api/users`

Is it possible to either override the the baseUrl defined in the connector, or replace a parameter from the request?

Thanks!

**Comment by Sammyjo20** [comment]
You could change the way you are making the requests by using the connector and then you could add a clientId constructor argument to the connector:

```php
<?php

$connector = new ApiConnector($clientId);
$request = new UserRequest;

$response = $connector->send($request);
```

Then inside your connector, you can write:

```php
class ApiConnector extends SaloonConnector
{
    public function defineBaseUrl(): string
    {
        return 'https://' . $this->clientId . '.host.com';
    }

    public function __construct(protected string $clientId)
    {
        //
    }
}
```

**Comment by Sammyjo20** [comment]
Alternatively, you could omit the base URL from the connector by writing an empty string ('') and then inside of your request you could write the base URL + endpoint.
